src/inmanta/env.py

```diff
@@ -44,6 +46,44 @@
 LOGGER = logging.getLogger(__name__)
 
 
+@dataclass
+class LocalPackagePath:
+    path: str
+    editable: bool = False
+
+
+class ProcessEnv:
+    """
+    Class to represent the Python environment this process is running in.
+    """
+    env_path: str = sys.executable
+
+    @classmethod
+    def install_from_source(cls, paths: List[LocalPackagePath]) -> None:
+        """
+        Install one or more packages from source. Any path arguments should be local paths to a package directory.
+        """
+        if len(paths) == 0:
+            raise Exception("install_from_source requires at least one package to install")
+        # make sure we only try to install from a local source
+        explicit_paths: Iterator[LocalPackagePath] = (
+            LocalPackagePath(path=os.path.join(".", path.path), editable=path.editable)
+            for path in paths
+        )
```

Comment:
From the comment I understand that we do this to ensure you cannot pass a URL to the command. Is that correct?

Reply:
A url or even the name of a package. This converts `inmanta-core` to `./inmanta-core`, making sure the command will fail if no such directory exists, instead of looking for it in the pip index.